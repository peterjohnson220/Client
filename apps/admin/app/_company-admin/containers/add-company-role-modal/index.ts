export * from './add-company-role-modal.component';
