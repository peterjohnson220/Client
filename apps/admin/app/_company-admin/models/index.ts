export * from './add-company-role-form.model';
