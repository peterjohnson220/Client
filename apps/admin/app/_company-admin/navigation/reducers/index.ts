import { createFeatureSelector, createSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromNavigationReducer from './navigation.reducer';

// Feature area state
export interface CompanyAdminNavigationState {
  navigation: fromNavigationReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  companyAdminNavigation: CompanyAdminNavigationState;
}

// Feature area reducers
export const reducers = {
  navigation: fromNavigationReducer.reducer,
};

// Select Feature Area
export const selectCompanyAdminNavigationState =
  createFeatureSelector<CompanyAdminNavigationState>('companyAdminNavigation');

// View Selectors
export const selectNavigationState =
  createSelector(selectCompanyAdminNavigationState, (state: CompanyAdminNavigationState) => state.navigation);

// Navigation Links
export const getLinks = createSelector(
  selectNavigationState, fromNavigationReducer.getLinks
);

export const getNavigationLinksLoading = createSelector(
  selectNavigationState, fromNavigationReducer.getLoading
);

export const getNavigationLinksLoadingError = createSelector(
  selectNavigationState, fromNavigationReducer.getLoadingError
);

