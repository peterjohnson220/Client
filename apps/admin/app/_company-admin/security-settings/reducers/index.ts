import { createFeatureSelector, createSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromPasswordSettingsReducer from './security-setting.reducer';

// Feature area state
export interface CompanyAdminPasswordSettingsState {
  passwordSettings: fromPasswordSettingsReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  companyAdminPasswordSettings: CompanyAdminPasswordSettingsState;
}

// Feature area reducers
export const reducers = {
  passwordSettings: fromPasswordSettingsReducer.reducer,
};

// Select Feature Area
export const selectCompanyAdminPasswordSettingsState =
  createFeatureSelector<CompanyAdminPasswordSettingsState>('companyAdminPasswordSettings');

// View Selectors

export const selectPasswordSettingsState =
  createSelector(selectCompanyAdminPasswordSettingsState, (state: CompanyAdminPasswordSettingsState) => state.passwordSettings);

// Password Settings
export const getPasswordSettings = createSelector(
  selectPasswordSettingsState, fromPasswordSettingsReducer.getPasswordSettings
);

export const getPasswordSettingsLoading = createSelector(
  selectPasswordSettingsState, fromPasswordSettingsReducer.getLoading
);

export const getPasswordSettingsLoadingError = createSelector(
  selectPasswordSettingsState, fromPasswordSettingsReducer.getLoadingError
);

export const getPasswordSettingsModalOpen = createSelector(
  selectPasswordSettingsState, fromPasswordSettingsReducer.getSaveModalOpen
);

export const getPasswordSettingsSaveRequest = createSelector(
  selectPasswordSettingsState, fromPasswordSettingsReducer.getSaveRequest
);

export const getPasswordSettingsSaving = createSelector(
  selectPasswordSettingsState, fromPasswordSettingsReducer.getSaving
);

export const getPasswordSettingsSavingError = createSelector(
  selectPasswordSettingsState, fromPasswordSettingsReducer.getSavingError
);
