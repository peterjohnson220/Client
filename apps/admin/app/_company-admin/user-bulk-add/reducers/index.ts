import { createFeatureSelector, createSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromUserBulkAddReducer from './user-bulk-add.reducer';

// Feature area state
export interface CompanyAdminUserBulkAddState {
  userBulkAddSettings: fromUserBulkAddReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  companyAdminUserBulkAddSettings: CompanyAdminUserBulkAddState;
}

// Feature area reducers
export const reducers = {
  userBulkAddSettings: fromUserBulkAddReducer.reducer,
};

// Select Feature Area
export const selectCompanyAdminUserBulkAddState =
  createFeatureSelector<CompanyAdminUserBulkAddState>('companyAdminUserBulkAddSettings');

// View Selectors
export const selectUserBulkAddState =
  createSelector(selectCompanyAdminUserBulkAddState, (state: CompanyAdminUserBulkAddState) => state.userBulkAddSettings);

// User Bulk Add Settings
export const getLoaderState = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getLoaderState
);

export const getWorksheetNames = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getWorksheetNames
);

export const getStoreFileNameInSession = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getStoreFileNameInSession
);

export const getStoreFileNameInSessionSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getStoreFileNameInSessionSuccess
);

export const getStoreFileNameInSessionError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getStoreFileNameInSessionError
);

export const getStoreAndMapBulkAddUsersInSession = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getStoreAndMapBulkAddUsersInSession
);

export const getStoreAndMapBulkAddUsersInSessionSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getStoreAndMapBulkAddUsersInSessionSuccess
);

export const getStoreAndMapBulkAddUsersInSessionError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getStoreAndMapBulkAddUsersInSessionError
);

export const getSaveUsers = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getSaveUsers
);

export const getSaveUsersSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getSaveUsersSuccess
);

export const getSaveUsersError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getSaveUsersError
);

export const getImportCount = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getImportCount
);

export const getImportCountSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getImportCountSuccess
);

export const getImportCountError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getImportCountError
);

export const getValidateHeaders = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateHeaders
);

export const getValidateHeadersSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateHeadersSuccess
);

export const getValidateHeadersError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateHeadersError
);

export const getValidateRequiredFields = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateRequiredFields
);

export const getValidateRequiredFieldsSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateRequiredFieldsSuccess
);

export const getValidateRequiredFieldsError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateRequiredFieldsError
);

export const getValidateEmails = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateEmails
);

export const getValidateEmailsSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateEmailsSuccess
);

export const getValidateEmailsError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateEmailsError
);


export const getValidatePasswords = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidatePasswords
);

export const getValidatePasswordsSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidatePasswordsSuccess
);

export const getValidatePasswordsError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidatePasswordsError
);


export const getValidateUserRoles = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateUserRoles
);

export const getValidateUserRolesSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateUserRolesSuccess
);

export const getValidateUserRolesError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getValidateUserRolesError
);

export const getDownloadDataFileWithErrors = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getDownloadDataFileWithErrors
);

export const getDownloadDataFileWithErrorsSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getDownloadDataFileWithErrorsSuccess
);

export const getDownloadDataFileWithErrorsError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getDownloadDataFileWithErrorsError
);

export const getCompanyPasswordLength = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getCompanyPasswordLength
);

export const getCompanyPasswordLengthSuccess = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getCompanyPasswordLengthSuccess
);

export const getCompanyPasswordLengthError = createSelector(
  selectUserBulkAddState, fromUserBulkAddReducer.getCompanyPasswordLengthError
);






