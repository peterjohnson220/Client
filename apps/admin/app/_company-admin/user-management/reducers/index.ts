import { createFeatureSelector, createSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromUserManagementReducer from './user-management.reducer';

// Feature area state
export interface CompanyAdminUserManagementState {
  userForm: fromUserManagementReducer.IUserState;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  companyAdminUserManagement: CompanyAdminUserManagementState;
}

// Feature area reducers
export const reducers = {
  userForm: fromUserManagementReducer.reducer
};

// Select Feature Area
export const selectCompanyAdminUserManagementState =
  createFeatureSelector<CompanyAdminUserManagementState>('companyAdminUserManagement');

// View Selectors
export const selectUserFormState =
  createSelector(selectCompanyAdminUserManagementState, (state: CompanyAdminUserManagementState) => state.userForm);


// User Form
export const getUserState = createSelector(
  selectUserFormState, fromUserManagementReducer.getUserState
);

export const getUserStateRoles = createSelector(
  selectUserFormState, fromUserManagementReducer.getRoles
);

export const getUserStateUser = createSelector(
  selectUserFormState, fromUserManagementReducer.getUser
);

export const getUserStateLoading = createSelector(
  selectUserFormState, fromUserManagementReducer.getUserLoading
);

export const getUserStateLoaded = createSelector(
  selectUserFormState, fromUserManagementReducer.getUserLoaded
);

export const getUserStateApiError = createSelector(
  selectUserFormState, fromUserManagementReducer.getUserApiError
);
