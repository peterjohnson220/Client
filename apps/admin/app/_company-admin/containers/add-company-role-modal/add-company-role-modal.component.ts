import { Component, ElementRef, OnDestroy, OnInit, ViewChild } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';

import { Store } from '@ngrx/store';
import { Observable, Subscription } from 'rxjs';
import { filter, take } from 'rxjs/operators';

import { UserAssignedRole, UserContext } from 'libs/models/security';
import { PfValidators } from 'libs/forms/validators';
import * as fromRootState from 'libs/state/state';

import { UserRoleValidationService } from '../../services';
import * as fromUserRoleViewReducer from '../../reducers';
import * as fromUserRoleActions from '../../actions/user-role-view.action';

@Component({
  selector: 'pf-add-company-role-modal',
  providers: [UserRoleValidationService],
  templateUrl: './add-company-role-modal.component.html',
  styleUrls: ['./add-company-role-modal.component.scss']
})
export class AddCompanyRoleModalComponent implements OnInit, OnDestroy {
  addCompanyRoleModalIsOpen$: Observable<boolean>;
  userContext$: Observable<UserContext>;
  userContext: UserContext;
  addCompanyRoleForm: FormGroup;
  AddCompanyRoleError: string;
  currentCompanyRoleName: string;
  formValueChangesSubscription: Subscription;
  addCompanyRoleFormSubscription: Subscription;
  addCompanyRoleErrorSubscription: Subscription;
  @ViewChild('companyRoleName') companyRoleName: ElementRef;

  constructor(private store: Store<fromUserRoleViewReducer.State>,
              private rootStore: Store<fromRootState.State>,
              private formBuilder: FormBuilder,
              public validationService: UserRoleValidationService) {
    this.userContext$ = store.select(fromRootState.getUserContext);
    this.addCompanyRoleModalIsOpen$ = this.store.select(fromUserRoleViewReducer.getAddCompanyRoleModalIsOpen);

    this.addCompanyRoleFormSubscription = this.store.select(fromUserRoleViewReducer.getAddCompanyRoleForm).subscribe(addCompanyRoleForm => {
      if (addCompanyRoleForm) {
        this.currentCompanyRoleName = addCompanyRoleForm.CompanyRoleName;
      }

    });

    this.addCompanyRoleErrorSubscription = this.store.select(fromUserRoleViewReducer.getAddCompanyRoleError)
      .subscribe(addCompanyRoleError => {
        this.AddCompanyRoleError = addCompanyRoleError;
      });

    this.initForm();
  }

  ngOnInit() {
    this.formValueChangesSubscription = this.addCompanyRoleForm.valueChanges.subscribe(() => {
      this.store.dispatch(new fromUserRoleActions.FieldChange(this.addCompanyRoleForm.value));
    });

    this.userContext$.pipe(
      filter(uc => !!uc),
      take(1)
    ).subscribe(uc => {
      this.userContext = uc;
    });
  }

  onAddCompanyRoleSubmit() {
    const newCompanyRole: UserAssignedRole = this.buildUserAssignedrole();
    this.store.dispatch(new fromUserRoleActions.AddCompanyRole(newCompanyRole));
  }

  onAddCompanyRoleDismiss() {
    this.store.dispatch(new fromUserRoleActions.CloseAddCompanyRoleModal());
  }

  initForm() {
    const validator = this.validationService;
    this.addCompanyRoleForm = this.formBuilder.group({
      CompanyRoleName: [
        '',
        [
          PfValidators.required,
          Validators.minLength(validator.NAME_LENGTH.min),
          Validators.maxLength(validator.NAME_LENGTH.max),
          validator.validateSpecialChars
        ]
      ]
    });
  }

  onFieldBlur(formControlName: string) {
    // when a text field is blurred trim it's contents
    const formControl = this.addCompanyRoleForm.controls[formControlName];
    if (formControl && formControl.value) {
      formControl.setValue(formControl.value.trim());
    }
  }

  ngOnDestroy() {
    this.formValueChangesSubscription.unsubscribe();
    this.addCompanyRoleFormSubscription.unsubscribe();
    this.addCompanyRoleErrorSubscription.unsubscribe();
  }

  private buildUserAssignedrole(): UserAssignedRole {
    return {
      RoleId: 0,
      CompanyId: this.userContext ? this.userContext.CompanyId : 0,
      RoleName: this.currentCompanyRoleName,
      IsSystemRole: false,
      Assigned: false,
      Permissions: [],
    };
  }
}
