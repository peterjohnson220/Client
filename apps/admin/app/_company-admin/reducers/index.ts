import { createFeatureSelector, createSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromUserRoleViewReducer from './user-role-view.reducer';
import * as fromUserRoleUserTabReducer from './user-role-users-tab.reducer';
import * as fromUserRoleFunctionTabReducer from './user-role-functions-tab.reducer';
import * as fromNavigationReducer from './navigation.reducer';
import * as fromDataAccessTabReducer from './role-data-access-tab.reducer';

// Feature area state
export interface UserRoleViewStateMain {
  userRoleView: fromUserRoleViewReducer.IUserRoleState;
  userRoleUserTab: fromUserRoleUserTabReducer.State;
  userRoleFunctionTab: fromUserRoleFunctionTabReducer.State;
  navigation: fromNavigationReducer.State;
  dataAccessTab: fromDataAccessTabReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  userRoleAdminMain: UserRoleViewStateMain;
}

// Feature area reducers
export const reducers = {
  userRoleView: fromUserRoleViewReducer.reducer,
  userRoleUserTab: fromUserRoleUserTabReducer.reducer,
  userRoleFunctionTab: fromUserRoleFunctionTabReducer.reducer,
  navigation: fromNavigationReducer.reducer,
  dataAccessTab: fromDataAccessTabReducer.reducer
};

// Select Feature Area
export const selectUserRoleAdminMainState =
  createFeatureSelector<UserRoleViewStateMain>('userRoleAdminMain');

// User Role View Selectors
export const selectUserRoleState =
  createSelector(selectUserRoleAdminMainState, (state: UserRoleViewStateMain) => state.userRoleView);

export const selectUserRoleFunctionTabState =
  createSelector(selectUserRoleAdminMainState, (state: UserRoleViewStateMain) => state.userRoleFunctionTab);

export const selectRoleDataAccessTabState =
  createSelector(selectUserRoleAdminMainState, (state: UserRoleViewStateMain) => state.dataAccessTab);

export const selectUserRoleUserTabState =
  createSelector(selectUserRoleAdminMainState, (state: UserRoleViewStateMain) => state.userRoleUserTab);

export const selectNavigationState =
  createSelector(selectUserRoleAdminMainState, (state: UserRoleViewStateMain) => state.navigation);

export const getUserRoleViewState = createSelector(
  selectUserRoleState, fromUserRoleViewReducer.getUserRoleViewState
);

export const getUserRoleCurrentTabState = createSelector(
  selectUserRoleState, fromUserRoleViewReducer.getUserRoleCurrentTab
);

export const getCurrentUserRole = createSelector(
  selectUserRoleState, fromUserRoleViewReducer.getCurrentUserRole
);

export const getCompanyRoles = createSelector(
  selectUserRoleState, fromUserRoleViewReducer.getCompanyRoles
);

export const getAddCompanyRoleModalIsOpen = createSelector(
  selectUserRoleState, fromUserRoleViewReducer.getAddCompanyRoleModalIsOpen
);

export const getAddCompanyRoleForm = createSelector(
  selectUserRoleState, fromUserRoleViewReducer.getAddCompanyRoleForm
);

export const getAddCompanyRoleError = createSelector(
  selectUserRoleState, fromUserRoleViewReducer.getAddCompanyRoleError
);

export const getRoleApiResponse = createSelector(
  selectUserRoleState, fromUserRoleViewReducer.getRoleApiResponse
);

export const getSaveButtonDisabled = createSelector(
  selectUserRoleState, fromUserRoleViewReducer.getSaveButtonDisabled
);

// Functions tab
export const getFunctionTabPermissions = createSelector(
  selectUserRoleFunctionTabState, fromUserRoleFunctionTabReducer.getFunctionTabPermissions
);

export const getFunctionTabPendingChanges = createSelector(
  selectUserRoleFunctionTabState, fromUserRoleFunctionTabReducer.getFunctionTabPendingChanges
);

export const getCurrentCheckedPermissionIds = createSelector(
  selectUserRoleFunctionTabState, fromUserRoleFunctionTabReducer.getCheckedPermissions
);

export const getCheckboxesDisabled = createSelector(
  selectUserRoleFunctionTabState, fromUserRoleFunctionTabReducer.getCheckboxesDisabled
);

// Users tab
export const getUsersAndRoles = createSelector(
  selectUserRoleUserTabState, fromUserRoleUserTabReducer.getUsersAndRoles
);

export const getUsersAndRolesError = createSelector(
  selectUserRoleUserTabState, fromUserRoleUserTabReducer.getUsersAndRolesError
);

export const getUsersInActiveRole = createSelector(
  selectUserRoleUserTabState, fromUserRoleUserTabReducer.getUsersInActiveRole
);

export const getUsersNotInActiveRole = createSelector(
  selectUserRoleUserTabState, fromUserRoleUserTabReducer.getUsersNotInActiveRole
);

export const getUsersTabHasPendingChanges = createSelector(
  selectUserRoleUserTabState, fromUserRoleUserTabReducer.getUsersTabHasPendingChanges
);

export const getUserIdsToSave = createSelector(
  selectUserRoleUserTabState, fromUserRoleUserTabReducer.getUserIdsToSave
);

// Navigation Links
export const getLinks = createSelector(
  selectNavigationState, fromNavigationReducer.getLinks
);

export const getNavigationLinksLoading = createSelector(
  selectNavigationState, fromNavigationReducer.getLoading
);

export const getNavigationLinksLoadingError = createSelector(
  selectNavigationState, fromNavigationReducer.getLoadingError
);

// Data Access Tab

export const getDataTypes = createSelector(
  selectRoleDataAccessTabState, fromDataAccessTabReducer.getDataTypes
);

export const getRoleDataRestrictions = createSelector(
  selectRoleDataAccessTabState, fromDataAccessTabReducer.getRoleDataRestrictions
);

export  const getDataAccessTabPendingChanges = createSelector(
  selectRoleDataAccessTabState, fromDataAccessTabReducer.getDataAccessTabHasPendingChanges
);
