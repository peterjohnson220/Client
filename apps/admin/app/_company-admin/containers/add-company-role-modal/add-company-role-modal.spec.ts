import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { combineReducers, Store, StoreModule } from '@ngrx/store';

import { PfFormsModule } from 'libs/forms';
import { generateMockUserAssignedRole } from 'libs/models/security';
import * as fromRootState from 'libs/state/state';

import { AddCompanyRoleModalComponent } from './add-company-role-modal.component';
import * as fromUserRoleViewReducer from '../../reducers';
import * as fromUserRoleActions from '../../actions/user-role-view.action';

describe('AddCompanyRoleModalComponent', () => {
  let component: AddCompanyRoleModalComponent;
  let fixture: ComponentFixture<AddCompanyRoleModalComponent>;
  let store: Store<fromUserRoleViewReducer.State>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          companyAdminMain: combineReducers(fromUserRoleViewReducer.reducers)
        }),
        PfFormsModule
      ],
      declarations: [AddCompanyRoleModalComponent],
    })
      .compileComponents();

    store = TestBed.get(Store);
    spyOn(store, 'dispatch');
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AddCompanyRoleModalComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch the CloseAddCompanyRoleModal action when the modal is dismissed', () => {
    component.onAddCompanyRoleDismiss();

    const dismissAction = new fromUserRoleActions.CloseAddCompanyRoleModal();
    expect(store.dispatch).toHaveBeenCalledWith(dismissAction);
  });

  it('should dispatch AddCompanyRole action when the modal is submitted', () => {
    const mockCompanyRole = generateMockUserAssignedRole();

    component.currentCompanyRoleName = mockCompanyRole.RoleName;
    component.onAddCompanyRoleSubmit();
    const submitAction = new fromUserRoleActions.AddCompanyRole(mockCompanyRole);
    expect(store.dispatch).toHaveBeenCalledWith(submitAction);
  });

  it('should trim entered text fields on blur', () => {
    const formControlNames = ['CompanyRoleName'];

    component.initForm();

    // set each form value to have a starting and trailing space
    formControlNames.forEach(controlName => {
      component.addCompanyRoleForm.controls[controlName].setValue(` ${controlName} `);
    });

    // call onFieldBlur for each control
    formControlNames.forEach(controlName => {
      component.onFieldBlur(controlName);
    });

    // loop through each of the controls, and assert the first and last values are not spaces
    formControlNames.forEach(controlName => {
      expect(component.addCompanyRoleForm.controls[controlName].value.charAt(0)).not.toMatch(' ');

      const lastCharacterIndex = component.addCompanyRoleForm.controls[controlName].value.length - 1;
      expect(component.addCompanyRoleForm.controls[controlName].value.charAt(lastCharacterIndex)).not.toMatch(' ');
    });
  });
});

