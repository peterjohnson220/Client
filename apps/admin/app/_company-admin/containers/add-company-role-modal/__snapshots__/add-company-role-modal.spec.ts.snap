// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCompanyRoleModalComponent should match snapshot 1`] = `
<pf-add-company-role-modal
  AddCompanyRoleError="undefined"
  addCompanyRoleErrorSubscription={[Function Subscriber]}
  addCompanyRoleForm={[Function FormGroup]}
  addCompanyRoleFormSubscription={[Function Subscriber]}
  addCompanyRoleModalIsOpen$={[Function Store]}
  companyRoleName={[Function ElementRef]}
  formBuilder={[Function FormBuilder]}
  formValueChangesSubscription={[Function Subscriber]}
  rootStore={[Function Store]}
  store={[Function Store]}
  userContext$={[Function Store]}
  validationService={[Function UserRoleValidationService]}
>
  <pf-modal-form
    ng-reflect-backdrop="true"
    ng-reflect-form-group="[object Object]"
    ng-reflect-is-open_="[object Object]"
    ng-reflect-modal-id="au-modal-add-company-role"
    ng-reflect-primary-button-text="Create"
    ng-reflect-reset-form-on-close="true"
    ng-reflect-size="md"
    ng-reflect-submitting="false"
    ng-reflect-title="Create New Role"
  >
    
    
    
    <div
      class="modal-container"
      id="au-modal-add-company-role"
    />
  </pf-modal-form>
</pf-add-company-role-modal>
`;
